network/f5/bigip_gtm_pool.py

```diff
@@ -0,0 +1,274 @@
+            state = dict(type='str', required=True, choices=['present', 'absent', 'enabled', 'disabled']),
+            pool = dict(type='str', required=True),
+            partition = dict(type='str', default='Common'),
+            virtual_server_server = dict(type='str', required=False),
+            virtual_server_name = dict(type='str', required=False),
+            lb_method = dict(type='str', required=True, choices=lb_method_choices)
+        ),
+        supports_check_mode=True
+    )
+
+    if not bigsuds_found:
+        module.fail_json(msg="the python bigsuds library is required")
+
+    server = module.params['server']
+    user = module.params['user']
+    password = module.params['password']
+    state = module.params['state']
+    partition = module.params['partition']
+    pool = "/%s/%s" % (partition, module.params['pool'])
+    virtual_server_name = module.params['virtual_server_name']
+    virtual_server_server = module.params['virtual_server_server']
+    lb_method = module.params['lb_method']
+
+    try:
+        api = bigip_api(server, user, password)
+
+        result = {'changed': False}  # default
+
+        if state == 'absent':
+            if pool and virtual_server_name and virtual_server_server:
+                if pool_exists(api, pool) and member_exists(api, pool, virtual_server_name, virtual_server_server):
+                    if not module.check_mode:
+                        remove_pool_member(api, pool, virtual_server_name, virtual_server_server)
+                        remove_pool(api, pool)
+                        result = {'changed': True}
+                    else:
+                        # check-mode return value
+                        result = {'changed': True}
+            elif pool_exists(api, pool):
+                if not module.check_mode:
+                    remove_pool(api, pool)
+                    result = {'changed': True}
+                else:
+                    # check-mode return value
+                    result = {'changed': True}
+        elif state == 'present':
+            update = False
+            if not pool_exists(api, pool):
+                if not module.check_mode:
+                    try:
+                        add_pool(api, pool, lb_method)
+                        result = {'changed': True}
+                    except bigsuds.OperationFailed, e:
+                        if "already exists" in str(e):
+                            update = True
+                        else:
+                            # genuine exception
+                            raise
+                    # else:
+                    # stub add attributes to pool just created here
+                else:
+                    # check-mode return value
+                    result = {'changed': True}
+            else:
+                # pool exists -- potentially modify attributes
+                update = True
+            # update attributes stub
+            # if update:
```

Comment:
Yeah I personally don't have a usecase for it and got lazy but stubbed out the basics. I could probably spend a bit of time and work it out (or remove the stub to prevent confusion).